networkx/algorithms/community/tests/test_modularity_max.py

```diff
@@ -24,6 +24,19 @@ def test_modularity_communities(func):
     assert set(func(G)) == expected
 
 
+def test_modularity_communities_weighted():
+    G = nx.balanced_tree(2, 3)
+    for (a, b) in G.edges:
+        if ((a == 1) | (a == 2)) & (b != 0):
```

Comment:
Just a nit, but I think logical operators should be used here instead of bitwise

```suggestion
        if ((a == 1) or (a == 2)) and (b != 0):
```